### DISCUSSION TITLE
Storybook for admin components

### DISCUSSION THREAD

**Discussion by msoyka** [payloadcms/payload#2165] [General]
Looking to add custom admin react components, but wanted to leverage the build in styles already created by payload. Is there a storybook or doc site that would show all available components?

Example:
```
<button className="btn btn--size-medium btn--style-primary btn--has-tooltip">
      Text
</button>
```

**Comment by denolfe** [comment]
We have a roadmap item for releasing some design assets. https://github.com/payloadcms/payload/discussions/1471

Figma is specifically named in that item. How important would be to have this somewhere beyond Figma?

**Reply by msoyka** [reply]
My main goal is trying to white-label the CMS admin while leveraging the premade payload react components to integrate custom components without having to rewrite new components, but would be great to have an environment that showed the props that were available and how they behaved. Similar to a ui-library docs site.

**Side note:** After listening to some of your youtube video tutorials, I came to the belief that your team is not a fan of leveraging ui frameworks, but would also be a great feature to turn off all styling and theming to allow for a complete overhaul of the admin with a UI-framework of your choice.

**Reply by jmikrut** [reply]
Hey @msoyka — we actually are planning on building a Payload component library! Basically exactly what you're asking for. 

I thought this was on the roadmap. It differs a bit from providing design resources, but we are definitely going to do this. I thought it was Priority 2 as well. 

I will have the team check through our roadmap items and ensure that this one is properly surfaced / logged. I agree 100% that it is needed.

PS - we don't use UI frameworks within our frontends, because they're typically highly branded and deliberate. But the Payload components are already built for a specific purpose and _**should**_ be reusable! So that's why I see this as a no-brainer.

more soon

**Reply by jacobsfletch** [reply]
> we actually are planning on building a Payload component library

Here's that roadmap item: https://github.com/payloadcms/payload/discussions/293
